### PR TITLE
Reorganise nats config and nats backend import paths

### PR DESCRIPTION
**Pull request by nachtmaar**
**Description**

Changes proposed in this pull request:

Preparation for PR https://github.com/kyma-project/kyma/issues/15705:

- Moves the `env.NatsConfig` to the `pkg/backend/nats` package as it is specific to the backend (nats in this case)
	- In the future we can host it under `pkg/backend/jetstreamv2` if `pkg/backend/nats` vanishes away
- For now we cannot host it directly under `pkg/backend/jetstreamv2` since it imports `pkg/backend/nats` (for the v1alpha1 -> v1alpha2 conversion). By moving it to `pkg/backend/nats` we make the config specific to the backend and avoid a cyclic import for now.
- Names the import path for `pkg/backend/nats` as `backendnats` in all files (makes it uniform)

**Related issue(s)**

See also https://github.com/kyma-project/kyma/issues/15705

### CODE REVIEW

**Review by friedrichwilken — CHANGES_REQUESTED** [review]
I like these changes. Clean up the nits please and I will approve this. :)

File: components/eventing-controller/pkg/backend/nats/config.go [hunk 1]
```diff
@@ -25,8 +28,7 @@ type NatsConfig struct {
 
 	// JetStream-specific configs
 	// Name of the JetStream stream where all events are stored.
-	JSStreamName string `envconfig:"JS_STREAM_NAME" required:"true"`
-	// Prefix for the subjects in the stream
```
[the remark this answers is not in the document]
Comment (nachtmaar) [reply components/eventing-controller/pkg/backend/nats/config.go:29]: By accident, will bring it back :)

**Review by friedrichwilken — APPROVED** [review]
/lgtm

**Pull request merged**